File: sub_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>

void chatterCallback(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("接收到数据: [%s]", msg->data.c_str());
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "listener");
    ros::NodeHandle n;
    ros::Subscriber sub = n.subscribe("/driver/encoder", 1000, chatterCallback);
    ros::spin();
    return 0;
}

File: sub_node.py
#!/usr/bin/env python3
# coding=utf-8
  

import rospy  
from std_msgs.msg import String  

def callback(data):  
    rospy.loginfo("接收到数据: %s", data.data)  

def listener():  
    rospy.init_node('listener', anonymous=True)  
    rospy.Subscriber("/driver/encoder", String, callback)  
    # rospy.Subscriber("/driver/evl", String, callback)
    # rospy.Subscriber("/driver/imv", String, callback)
    # rospy.Subscriber("/driver/mag", String, callback)
    # rospy.Subscriber("/driver/scan", String, callback)
    rospy.spin()  


if __name__ == '__main__':  
    listener()
